dask/dataframe/io/parquet/core.py

```diff
@@ -0,0 +1,558 @@
+        Whether or not to write the index. Defaults to True.
+    append : bool, optional
+        If False (default), construct data-set from scratch. If True, add new
+        row-group(s) to an existing data-set. In the latter case, the data-set
+        must exist, and the schema must match the input data.
+    ignore_divisions : bool, optional
+        If False (default) raises error when previous divisions overlap with
+        the new appended divisions. Ignored if append=False.
+    partition_on : list, optional
+        Construct directory-based partitioning by splitting on these fields'
+        values. Each dask partition will result in one or more datafiles,
+        there will be no global groupby.
+    storage_options : dict, optional
+        Key/value pairs to be passed on to the file-system backend, if any.
+    write_metadata_file : bool, optional
+        Whether to create the special "_metadata" file,
+    compute : bool, optional
+        If True (default) then the result is computed immediately. If False
+        then a ``dask.delayed`` object is returned for future computation.
+    **kwargs
+        Extra options to be passed on to the specific backend.
+
+    Examples
+    --------
+    >>> df = dd.read_csv(...)  # doctest: +SKIP
+    >>> dd.to_parquet(df, '/path/to/output/', compression='snappy')  # doctest: +SKIP
+
+    See Also
+    --------
+    read_parquet: Read parquet data to dask.dataframe
+    """
+    from dask import delayed
+
+    partition_on = partition_on or []
+    if isinstance(partition_on, string_types):
+        partition_on = [partition_on]
+
+    if set(partition_on) - set(df.columns):
+        raise ValueError(
+            "Partitioning on non-existent column. "
+            "partition_on=%s ."
+            "columns=%s" % (str(partition_on), str(list(df.columns)))
+        )
+
+    if compression != "default":
+        kwargs["compression"] = compression
+    elif "snappy" in compress:
+        kwargs["compression"] = "snappy"
+
+    if isinstance(engine, str):
+        engine = get_engine(engine)
+
+    fs, _, _ = get_fs_token_paths(
+        path, mode="wb", storage_options=storage_options
+    )
+    # Trim any protocol information from the path before forwarding
+    # ideally, this should be done as a method of the file-system
+    path = infer_storage_options(path)["path"]
+
+    # Save divisions and corresponding index name
+    # TODO: What if the division are not along the index?
+    division_info = {"divisions": df.divisions, "name": df.index.name}
+    if division_info["name"] == None:
+        division_info["name"] = "index"
+
+    # If write_index==True (default), reset the index and record the
+    # name of the original index in `index_cols` (will be `index` if None).
+    # `fastparquet` will use `index_cols` to specify the index column(s)
+    # in the metadata.  `pyarrow` will revert the `reset_index` call
+    # below if `index_cols` is populated (because pyarrow can handle
```

Comment:
but note that this will never give the efficient metadata-only range index. I guess we are generally giving up on the idea of using that optimisation.